### PR TITLE
Fix compilation in typescript 2.1

### PR DESCRIPTION
**Pull request by rictic**

### CODE REVIEW

File: runner/config.ts [hunk 1]
```diff
@@ -436,17 +436,20 @@ export async function expand(context: Context): Promise<void> {
 function expandDeprecated(context: Context) {
   const options = context.options;
   // We collect configuration fragments to be merged into the options object.
-  const fragments: {plugins: {sauce: {}, local?: {}}}[] = [];
+  const fragments = [];
```
Comment (rictic) [inline runner/config.ts]: This can now be properly inferred.

File: runner/config.ts [hunk 2]
```diff
@@ -436,17 +436,20 @@ export async function expand(context: Context): Promise<void> {
...
   browsers = <any>_.compact(<any>browsers);
   if (browsers.length > 0) {
     context.emit(
-        'log:warn', 'The --browsers flag/option is deprecated. Please use ' +
+        'log:warn',
+        'The --browsers flag/option is deprecated. Please use ' +
             '--local and --sauce instead, or configure via plugins.' +
             '[local|sauce].browsers.');
-    const fragment = {plugins: {sauce: {}, local: {}}};
+    const fragment: {plugins: {[name: string]: {browsers?: Browser[]}}} = {
```
Comment (rictic) [inline runner/config.ts]: This now needs to be told that plugins can have arrays of browsers.

**Comment by justinfagnani** [comment]
Shouldn't this include a bump of the typescript version to ^2.1.0?

**Comment by rictic** [comment]
Good point, done

File: runner/config.ts (hunk 1 above)
Comment (justinfagnani) [inline runner/config.ts]: That's a little much inference, IMO, but hey. I hope they're keeping a spec.

**Pull request merged**